media_driver/agnostic/media_srcs.cmake

```diff
@@ -85,8 +88,103 @@ if (BUILD_KERNELS)
+        if(value_str STREQUAL "") # old style version
+            message(FATAL_ERROR "Failed to find macro ${name} in the file: ${file}")
+        endif()
+        string(REPLACE "#define ${name} " "" value_str ${value_str})
+        set(${value} ${value_str} PARENT_SCOPE)
+    endfunction()
+
+    # Function generates kernel for the specified platform. It assumes that generated
+    # kernel should be placed in the certain directory (see ${krn_dir}).
+    function(gen_kernel_from_cm name platform index sources)
+        platform_to_genx(${platform} genx kind)
+
+        set(krn ig${name}krn_g${genx})
+        set(krn_dir ${CMAKE_SOURCE_DIR}/media_driver/agnostic/${platform}/codec/kernel_free)
+        set(out_dir "${CMAKE_CURRENT_BINARY_DIR}/kernels/codec/${platform}")
+
+        add_custom_command(
+            OUTPUT ${out_dir}
+            COMMAND ${CMAKE_COMMAND} -E make_directory ${out_dir}
+            COMMENT "Create codec kernels output directory: ${out_dir}")
+
+        # Compiling all the given sources
+        set(dats "")
+        if(genx STREQUAL "11")
+          set(genx "11lp")
+        endif()
+        foreach(src ${sources})
+            get_cm_dat_objs(${src} objs) # there are other otputs from cmc command, but we use only .dat
+            add_custom_command(
+                OUTPUT ${objs}
+                DEPENDS ${out_dir} ${src}
+                WORKING_DIRECTORY ${out_dir}
+                COMMAND ${CMC}
+                    -c -Qxcm -Qxcm_jit_target=gen${genx}
+                    -mCM_emit_common_isa -mCM_no_input_reorder -mCM_jit_option="-nocompaction"
+                    ${src})
+            set(dats ${dats} ${objs})
+        endforeach()
+
+        # Generating source from the .krn file
+        get_c_macro_int(${CMAKE_CURRENT_LIST_DIR}/common/codec/kernel/codeckrnheader.h
+            "IDR_CODEC_TOTAL_NUM_KERNELS" IDR_CODEC_TOTAL_NUM_KERNELS)
+        add_custom_command(
+            OUTPUT ${krn_dir}/${krn}.c ${krn_dir}/${krn}.h
+            DEPENDS KernelBinToSource ${CMAKE_CURRENT_LIST_DIR}/common/codec/kernel/merge.py ${dats}
+            WORKING_DIRECTORY ${out_dir}
+            COMMAND ${PYTHON} ${CMAKE_CURRENT_LIST_DIR}/common/codec/kernel/merge.py -o ${krn}.krn ${dats}
+            # ${index} is needed to match a description in the following file:
+            #   media_driver/agnostic/common/codec/kernel/codeckrnheader.h
+            COMMAND KernelBinToSource -i ${krn}.krn -o ${krn_dir}/ -v ${krn} -index ${index} -t ${IDR_CODEC_TOTAL_NUM_KERNELS}
+            COMMENT "Generate source file from krn")
+    endfunction()
+
+    # List of kernel sources to build.
+    # NOTE: Order is important!! It should match the order in which sub-kernels are described
+    # in the corresponding strcuture in the driver. For example, HME kernel should
+    # match HmeDsScoreboardKernelHeaderG11
+    list(APPEND HME_KRN_SOURCES
+        ${CMAKE_CURRENT_LIST_DIR}/gen11/codec/kernel_free/Source/downscale_kernel_genx.cpp
```

Comment:
I don't see that IDR_CODEC_HME_DS_SCOREBOARD_KERNEL kernel is used anywhere else except gen11, so I think we can left these sources here for now.